pkg/adapter/containers.go

```diff
@@ -103,3 +103,29 @@ func (r *LocalRuntime) KillContainers(ctx context.Context, cli *cliconfig.KillVa
 	}
 	return ok, failures, nil
 }
+
+// RemoveContainers removes container(s) based on CLI inputs.
+func (r *LocalRuntime) RemoveContainers(ctx context.Context, cli *cliconfig.RmValues) ([]string, map[string]error, error) {
+	var (
+		ok       = []string{}
+		failures = map[string]error{}
+	)
+
+	ctrs, err := shortcuts.GetContainersByContext(cli.All, cli.Latest, cli.InputArgs, r.Runtime)
+	if err != nil {
+		// Force may be used to remove containers no longer found in the database
+		if cli.Force && len(cli.InputArgs) > 0 && errors.Cause(err) == libpod.ErrNoSuchCtr {
+			r.RemoveContainersFromStorage(cli.InputArgs)
+		}
+		return ok, failures, err
+	}
+
+	for _, c := range ctrs {
```

Comment:
for the local client, we want the parallel functions to still be used. it was a significant complaint from the field and makes a huge difference in execution times.  For the remote client, it is a nth but dealer's choice.  for the record, i have not been parallelizing remote calls.

Reply:
@baude Okay, now that I have a number of these under my belt, I'll put the parallel code in.